src/app/core/store/effects.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { ofType, Actions, createEffect } from '@ngrx/effects';
+import { map, switchMap, mergeMap, filter, tap } from 'rxjs/operators';
+import { of, from } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { AuthServices } from '../../shared/auth.srevices';
+
+import { User } from '../interfaces';
+import {
+  ActionTypes,
+  LoginAction,
+  LoginSuccessAction,
+  LoginFailedAction,
+  RegisterSuccessAction,
+} from '../store/action';
+
+@Injectable()
+export class Effects {
+  constructor(
+    private actions$: Actions,
+    private http: HttpClient,
+    private auth: AuthServices
```

Comment:
do not use httpClient in effect, API request is service responsibility 
```suggestion
    private actions$: Actions,
    private auth: AuthServices
```